### ISSUE TITLE
Migrate US server to new provider

### ISSUE THREAD

**Issue by luisramos0** [openfoodfoundation/ofn-install#646]
## Description
As a result of the massive growth in use experienced this spring, both the OFN USA server and the OFN CAN server are at capacity, and moving to a bigger server is highly recommended. Both OFN CAN and OFN USA would like to move to servers that are run by more ethical and/or environmentally responsible companies. OFN CAN uses Digital Ocean, which is near the bottom of the list of environmentally responsible practices for web hosting services.

The USA's access for website and postgres/Zapier updates has become very difficult. We need to move to a server which can cope with the growth that OFN USA is experiencing.

### Pre-Migration

- [x] Buy a new server
- [x] Create a temporary subdomain and point it at the server
- [x] Set up some temporary configs for the new server (`hosts` entry, `host_vars`, etc)
- [x] Run the`setup` playbook on the new server
- [x] Use the new `letsencrypt_proxy` playbook on the current server, specifying the new server's IP
- [x] Run the`provision` playbook on the new server
- [x] Deploy the current release to the new server
- [x] Run the `db_integrations` playbook on the new server, if it uses integrations like Zapier.

### Migration

- [x] Set the current server in maintenance mode, with a nice custom message
- [x] Stop `unicorn` and `delayed_job` on the current server
- [x] Transfer the assets to the new server (includes all images, which are hosted locally on the US server) < -- is this using `playbooks/transfer_assets.yml`? 
- [x] Migrate the database to the new server with `db_transfer` playbook
- [x] Restart `delayed_job` and `unicorn` on the new server to ensure changes are picked up
- [x] Check everything is okay on the site
- [x] Run `transparent_proxy` playbook on the current server, pointing to the new one

### Post-Migration

- [x] Switch the domain name to point to the new server's IP
- [x] Leave some time for the DNS to propagate, then shut down the old server
- [x] Clean up the ofn-install inventory entries so only one remains, with the new IP address

### Optional Extras

- [x] Remove the temporary subdomain
- [x] Regenerate the certificate on the new server, with the temporary subdomain removed from the list https://github.com/openfoodfoundation/openfoodnetwork/issues/5418#issuecomment-629705179

## Expected Behavior
New server for USA can handle growth, zapier integrations and be in a vlues aligned provider.  

## Possible Fix
This is related to
https://community.openfoodnetwork.org/t/switch-to-green-hosting-when-new-projects-provisionning/1616

**Comment by lauriewayne** [comment]
Thank you for creating this issue, Luis! What communication is needed with Jim, the holder of the AWS account where the US server currently resides?

**Comment by luisramos0** [comment]
I dont think we need anything, we do have admin access to the server so we can take whatever we need from there.
He will probably just need to stop the instance **once the migration is done**. Which is really just to press a button.

**Comment by luisramos0** [comment]
I just confirmed the DNS is on AUS godaddy account (I have delegate access given by Kirsten to this godaddy account if anyone needs to change something).

**Comment by lauriewayne** [comment]
Great, thanks! Sometimes it's hard to get ahold of Jim. Will anything on our side break or be problematic if he doesn't press the button to stop the instance? If we will be impacted by his stopping or not stopping the instance, I should start working on that now.

**Comment by luisramos0** [comment]
no not at all. Stopping the instance means to switch it down when it's all migrated. The only problem of not switching down is that he will have to pay for it afterwards even if we dont use it.
This is something for **after the migration**.

**Comment by lauriewayne** [comment]
Perfect thank you @luisramos0! I will let him know after we are happily up and running on the new server.

**Comment by Matt-Yorkley** [comment]
Server migration template here: https://github.com/openfoodfoundation/openfoodnetwork/issues/5418 :wink:

**Comment by andrewpbrett** [comment]
I started a conversation with the current holder of the AWS account to start figuring out steps to transfer to another AWS machine. 

For later on, looking at the [specs on some of the other instances](https://github.com/openfoodfoundation/ofn-install/wiki/Servers-and-domains), most have 4 x CPU and 8GB memory. 

- On AWS EC2 that translates to an a1.xlarge reserved instance, which is $47/month (hosted in Oregon)
- On Digital Ocean a standard droplet with 4x CPU and 8GB memory is $40/month
- On Infomaniak an unmanaged cloud server with 4x CPU and 12GB memory is 49€/month (currently about $58)

Infomaniak does have the advantage of being based in Switzerland (privacy wise) and looks more values-aligned, not to mention the slightly higher specs.

**Comment by luisramos0** [comment]
Hey @andrewpbrett nice start!

>> On Infomaniak an unmanaged cloud server with 4x CPU and 12GB memory is 49€/month (currently about $58)

Are the servers in CH as well? The users will be in the US, you need a server in some US data center, right?


OVH, the  provider for France is cheap 
![image](https://user-images.githubusercontent.com/1640378/90621128-392d2900-e20b-11ea-982d-a56d95208caa.png)
I think they will have servers in the US.

**Comment by andrewpbrett** [comment]
OVH looks nice! They do have servers in the Eastern US (Virginia) as well as Montreal. Maybe Montreal would be appealing since it's close enough for latency but outside the scope of the Patriot Act etc.? (I don't know how much the physical location of the server makes a difference for that though, as opposed to how and where the provider is organized).

**Comment by lauriewayne** [comment]
Yay! I definitely prioritize reliability and customer service/reputation but given those, cheaper (and more ethical?) is awesome!!!  Thank you @luisramos0 and @andrewpbrett for your sleuthing!

**Comment by andrewpbrett** [comment]
I'm still waiting to get what we need from our current AWS provider for this to proceed. I'm hopeful that once we have it, we could first do a quick(er) transfer to an AWS account that we control and then later switch providers (probably to OVH, but need a little more research on it first).

**Comment by Matt-Yorkley** [comment]
Ok, so we're scheduled for migrating on Monday morning. I'm just doing a few checks now, I think there's a minor thing with the TLS certificates setup that needs a slight adjustment :ok_hand:

**Comment by Matt-Yorkley** [comment]
Alright, I think we're done. That was faaast! :rocket: 

I ran into a small email issue related to https://github.com/openfoodfoundation/ofn-install/issues/696, but it should be fixed now.

@andrewpbrett you can update the DNS now so the primary domain points to the new server's IP address. Note: we should leave the old server running for a day or so before shutting it down, and make sure nobody provisions the old server until then.

**Comment by Matt-Yorkley** [comment]
I think the total downtime was around 10 minutes :muscle:

**Comment by lauriewayne** [comment]
Oooh hey, email confirmations for new registrations want to go to"demo.openfoodnetwork.org" and the site url was set to that in the general settings. I changed the general settings to "openfoodnetwork.net" but confirmations still want to go to the demo site. Is there anything else that might be impacted by this (for example order confirmations yikes)?

**Comment by RachL** [comment]
ping @openfoodfoundation/core-devs some configurations are missing ☝️

**Comment by andrewpbrett** [comment]
Thanks for noticing that and reporting @lauriewayne. I redeployed the latest release after the site URL was changed and the emails now have the correct link. I think I mistakenly assumed that the values in general settings would have been carried over to the new site with the database, but thinking about that more I wouldn't be surprised if some of them are actually environment variables.... probably worth adding a warning/step in the migration "template"?
